src/frontend/src/pages/part/PartDetail.tsx

```diff
@@ -878,6 +984,40 @@ export default function PartDetail() {
               setTreeOpen(true);
             }}
             actions={partActions}
+            detail={
+              enableRevisionSelection ? (
+                <Stack gap="xs">
+                  <Text>{t`Select Part Revision`}</Text>
+                  <Select
```

Comment:
Is there a reason why you do all that manually? Wouldn't a StandaloneField in combination with the RelatedModelField work?

Reply:
Because we construct the list of options locally, (rather than fetch directly from the field) and then pass to the select, it won't quite work (without some refactoring of the `RelatedModelField`

Reply:
Ok, makes sense. Was just thinking from a Quick Look that the endpoint you fetch is a simple related model endpoint which the related model field could handle.